app/src/main/java/org/fossasia/openevent/app/data/LoginModel.java

```diff
@@ -43,18 +48,20 @@ public LoginModel(IUtilModel utilModel, EventService eventService, IDatabaseRepo
 
         return eventService
                 .login(new Login(username, password))
-                .doOnNext(loginResponse -> utilModel.saveToken(loginResponse.getAccessToken()))
                 .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .doOnComplete(() ->
-                    databaseRepository.getAllItems(User.class)
-                    .any(user -> !user.getEmail().equals(username))
-                    .filter(differentUser -> differentUser)
-                    .subscribeOn(Schedulers.io())
-                    .subscribe(differentUser ->
-                        utilModel.deleteDatabase()
-                            .subscribe()
-                    ));
+                .observeOn(AndroidSchedulers.mainThread());
+    }
+
+    @Override
+    public Observable<Boolean> isPreviousUser() {
+        return Observable.combineLatest(eventService.getUser(getAuthorization()),
+            databaseRepository.getAllItems(User.class).switchIfEmpty(Observable.just(new User(""))).firstElement().toObservable(), this::areSameUsers)
+            .subscribeOn(Schedulers.io())
```

Comment:
@iamareebjamal hey I think it would better to keep them separate. `LoginModel::login` should just handle login ie getting access token. Then before starting main activity it is presenter's work to check for data consistency and take actions accordingly. So I think current approach is fine. What do you say? doing all the stuff in the login itself will increase complexity.